pkg/store/cache/caching_bucket_factory.go

```diff
@@ -60,5 +95,36 @@ func NewCachingBucketFromYaml(yamlContent []byte, bucket objstore.Bucket, logger
 		return nil, errors.Errorf("unsupported cache type: %s", config.Type)
 	}
 
-	return NewCachingBucket(bucket, c, config.CachingBucketConfig, logger, reg)
+	cfg := NewCachingBucketConfig()
+
+	// Configure cache.
+	cfg.CacheGetRange("chunks", c, isTSDBChunkFile, config.ChunkSubrangeSize, config.ChunkObjectSizeTTL, config.ChunkSubrangeTTL, config.MaxChunksGetRangeRequests)
+	cfg.CacheExists("metafile", c, isMetaFile, config.MetafileExistsTTL, config.MetafileDoesntExistTTL)
+	cfg.CacheGet("metafile", c, isMetaFile, config.MetafileContentTTL, config.MetafileExistsTTL, config.MetafileDoesntExistTTL)
+
+	// Cache Iter requests for root.
+	cfg.CacheIter("blocks-iter", c, isBlocksRootDir, config.BlocksIterTTL, JsonIterCodec{})
+
+	cb, err := NewCachingBucket(bucket, cfg, logger, reg)
+	if err != nil {
+		return nil, err
+	}
+
+	return cb, nil
+}
+
+var chunksMatcher = regexp.MustCompile(`^.*/chunks/\d+$`)
+
+func isTSDBChunkFile(name string) bool { return chunksMatcher.MatchString(name) }
+
+func isMetaFile(name string) bool {
+	return strings.HasSuffix(name, metaFilenameSuffix) || strings.HasSuffix(name, deletionMarkFilenameSuffix)
```

Comment:
I feel like `metaFilenameSuffix` and `deletionMarkFilenameSuffix` are used only once... worth to maybe inline those here? :thinking:

Reply:
I suppose Go compiler is smart enough not to do concatenation on each call.

Reply:
Yes, I know, I am just seeing this as reader complexity. Human reading this code has to jump 100 lines to ensure something for no reason (variable not reused)